indico/modules/events/sessions/controllers/display.py

```diff
@@ -79,7 +85,42 @@ def _process(self):
                          'text/calendar')
 
 
-class RHExportSessionTimetableToPDF(RHDisplaySessionBase):
+class RHExportSessionTimetableToPDF(RHDisplaySessionBase, RHTimetableExportPDF):
     def _process(self):
-        pdf = get_session_timetable_pdf(self.session)
-        return send_file('session-timetable.pdf', BytesIO(pdf.getPDFBin()), 'application/pdf')
+        now = now_utc()
+        css = render_template('events/timetable/pdf/timetable.css')
+        event = self.event
+        entries = filter(
+            lambda e: self.session.id == e.id or (
+                e.session_block and self.session.id == e.session_block.session.id
+            ), get_nested_timetable(event)
+        )
```

Comment:
`filter(lambda)` is not very pythonic - use a list comprehension instead:

```python
        entries = [
            e for e in get_nested_timetable(event)
            if e.id == self.session.id or (e.session_block and self.session.id == e.session_block.session.id)
        ]
```

However, `self.session.id == e.id` looks quite fragile. What if there's let's say a contribution/break and a session with the same ID? Especially on small instances that can very easily happen for the first few events!

Reply:
We only allow session blocks to be exported so I have now changed it to account for what you said:

```python
        entries = [
            e for e in get_nested_timetable(event)
            if e.type == TimetableEntryType.SESSION_BLOCK and (
                e.id == self.session.id or (e.session_block and self.session.id == e.session_block.session.id)
            )
        ]
        ```